Mathlib/Tactic/Simproc/Divisors.lean

```diff
@@ -0,0 +1,48 @@
+import Mathlib.NumberTheory.Divisors
+import Mathlib.Lean.ToExpr
+import Mathlib.Data.Finset.Sort
+import Lean
+
+open Lean Meta
+
+simproc_decl Nat.divisorsEq (Nat.divisors _) := fun e => do
+  unless e.isAppOfArity `Nat.divisors 1 do return .continue
+  let some n ← fromExpr? e.appArg! | return .continue
+  --TODO: do we want to define `Nat.divisorsList` and use this instead of `n.divisors.sort (· ≤ ·)`?
+  let rhs ← (n.divisors.sort (· ≤ ·)).toFinsetExpr (α := ℕ)
+  /- The last two lines can be replace by `return .done {expr := rhs }` since the proof is `rfl`.
+  Not sure what's best here. -/
+  let pf ← Meta.mkDecideProof (← mkEq e rhs)
+  return .done {expr := rhs, proof? := pf }
```

Comment:
I have a vague memory that having the proof as rfl either doesn't work or is slower, but I don't quite remember...

Reply:
Yes, the proof by decide should usually be faster than the proof by rfl

Reply:
Is that easy to show as the case here?

Reply:
Fair point. It should be easy to check my guess!

Reply:
My guess would be that decidable equality of finsets is $O(n^2)$, but unfolding in the rfl is $O(n)$.

Reply:
Though the best-case equality decision on finsets is also $O(n)$, but I'm not sure if that's when the orders coincide or when they're reversed.

Reply:
I've changed the proof terms to `rfl` - are we happy to work with that?

Reply:
If the proof is rfl (and it's faster), this should be a dsimproc

Reply:
I've redone some measurements, and it seems like `rfl` is actually slower than `decide` so I've switched back to what we had originally.